typespec/Applications.Core/environments.tsp

```diff
@@ -103,24 +110,47 @@ model SecretConfig {
   secret?: string;
 }
 
-@doc("The Cloud providers configuration")
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
```

Comment:
I know we already do this in other places in our typespec codebase without any comments, but could you add a comment here mentioning that this is needed for `Record<unknown>`